src/ReplicatedStorage/Client/Controllers/InterfaceController.lua

```diff
@@ -0,0 +1,163 @@
+	then the interface will be mounted to a new SurfaceGui.
+
+	If the interface is already mounted, then this function will return the
+	`UnmountKey` that was previously returned by `Mount()`.
+]=]
+function InterfaceController:Mount(interface: GuiBase | GuiBase2d, parent: Instance?): UnmountKey
+	-- Check if interface is already mounted, if it is then return the unmount key
+	local storedUnmountKey = self._interfaceToUnmountKey[interface]
+	if storedUnmountKey then
+		return storedUnmountKey
+	end
+
+	-- Get/create parent gui if it doesn't exist
+	local parentGui
+
+	if parent ~= nil then
+		if parent:IsA("BasePart") or parent:IsDescendantOf(workspace) then
+			parentGui = if interface:IsA("SurfaceGui") then interface else Instance.new("SurfaceGui")
+			parentGui.ZIndexBehavior = Enum.ZIndexBehavior.Sibling
+			parentGui.Adornee = parent
+			parentGui.Parent = self.PlayerGui
+		else
+			parentGui = interface
+			parentGui.Parent = parent
+		end
+	else
+		parentGui = interface
+
+		if interface:IsA("ScreenGui") then
+			parentGui.Parent = self.PlayerGui
+		else
+			parentGui.Parent = self.GameGui
+		end
+	end
+
+	-- Mount the interface to the parent gui if we created a new one
+	if parentGui ~= interface then
+		interface.Parent = parentGui
+	end
+
+	-- Store the unmount key and function
+	local unmountKey = newproxy()
+
+	self._unmountKeyToParentGui[unmountKey] = interface
+	self._interfaceToUnmountKey[interface] = unmountKey
+	self._unmountFunctions[interface] = function()
+		self._unmountFunctions[interface] = nil
+		parentGui:Destroy()
+	end
+
+	-- Return the stored unmount key
+	return unmountKey :: UnmountKey
+end
+
+--[=[
+	Unmounts an interface that was previously mounted using `Mount()`. The
+	unmount process will be performed immediately, and will result in the
+	destruction of the interface. If you explicitly supplied a `parent` argument
+	to `Mount()`, the `parent` object will not be destroyed.
+
+	Returns a boolean indicating whether or not the interface was successfully
+	unmounted. If an error occurs, then the error message will be returned after
+	the boolean.
+]=]
+function InterfaceController:Unmount(unmountKey: UnmountKey): (boolean, ...any)
+	-- Get the interface to unmount
+	local mountedParentGui = self._unmountKeyToParentGui[unmountKey]
+	local mountedInterface = self._unmountKeyToInterface[unmountKey]
+	self._unmountKeyToParentGui[unmountKey] = nil
+	self._unmountKeyToInterface[unmountKey] = nil
+
+	-- Cleanup extra references
+	if mountedInterface ~= nil then
+		self._interfaceToUnmountKey[mountedInterface] = nil
+	end
+
+	-- Guard
+	if mountedParentGui == nil then return false end
+
+	-- Get the unmount function
+	local unmountFunction = self._unmountFunctions[mountedParentGui]
+	self._unmountFunctions[mountedParentGui] = nil
+
+	-- Guard
+	if unmountFunction == nil then return false end
+
+	-- Unmount the interface
+	return pcall(unmountFunction)
```

Comment:
Why are we silencing this error? Errors should not pass silently unless explicitly silenced - so I'm going to need good reasoning to understand this pcall.

Reply:
I wanted there to be a boolean representation for whether or not the unmounting was successful.

That being said, the only way the pcall would fail is if the parent gui doesn't exist.. which should be impossible based on the guard statements. I think it might just be better to not have a pcall and return true.